src/Compilers/CSharp/Test/IOperation/IOperation/IOperationTests_ILockStatement.cs

```diff
@@ -378,6 +400,262 @@ public void M()
             VerifyOperationTreeAndDiagnosticsForTest<LockStatementSyntax>(source, expectedOperationTree, expectedDiagnostics);
         }
 
+        [Fact, CompilerTrait(CompilerFeature.IOperation)]
+        public void ILockStatement_LockObject()
+        {
+            var source = """
+                using System;
+                using System.Threading;
+
+                Lock l = new Lock();
+                /*<bind>*/lock (l)
+                {
+                }/*</bind>*/
+                """;
+
+            var expectedOperationTree = """
+                ILockOperation (OperationKind.Lock, Type: null) (Syntax: 'lock (l) ... }')
+                  Expression:
+                    ILocalReferenceOperation: l (OperationKind.LocalReference, Type: System.Threading.Lock) (Syntax: 'l')
+                  Body:
+                    IBlockOperation (0 statements) (OperationKind.Block, Type: null) (Syntax: '{ ... }')
+                """;
+
+            var expectedDiagnostics = DiagnosticDescription.None;
+
+            VerifyOperationTreeAndDiagnosticsForTest<LockStatementSyntax>([source, LockTypeDefinition], expectedOperationTree, expectedDiagnostics);
+        }
+
+        [Fact, CompilerTrait(CompilerFeature.IOperation)]
+        public void ILockStatement_LockObject_Conditional()
+        {
+            var source = """
+                using System;
+                using System.Threading;
+
+                bool b = true;
+                Lock l1 = new Lock();
+                Lock l2 = new Lock();
+                /*<bind>*/lock (b ? l1 : l2)
```

Comment:
>b ? l1 : l2 [](http://example.com/codeflow?start=32&length=11)

Conditional test are interesting only for the graph related tests (we need to test control flow in the value) and only when the result type is the `Lock` type. You can certainly keep other tests, but, I do not find them interesting from the point of view of testing IOperation support.